### DISCUSSION TITLE
Can one explain to CFWheels that "" != null?

### DISCUSSION THREAD

**Discussion by adamcameron** [wheels-dev/wheels#1268] [General]
I have a table with a not-null column, and just tried to save an empty string to it. I got an error along the lines of "mycolumn can't be empty". Well, CFWheels, _yes it can_, it just can't be null.

I had a look in the issue history and found this: https://github.com/cfwheels/cfwheels/issues/478. That's a pretty lamentable state of affairs, but it's pretty old, so maybe things have improved.

Is there any way to get CFWheels **1.4.x** (sorry, but... it is what it is) to let empty strings be saved in not-null columns? Like specify an override for the validation of that property which I can hand-crank a validation handler to not baulk at empty strings?

**Comment by smilebringergames** [comment]
If you're running a lower version of Wheels (like I am as well), you just
end up modifying parts of your Wheels functions (and never updating them)
or you end up solving issues like these with checking for an empty string
to save a not-empty space in those fields.
" " Is not null. "" Is null.

**Comment by chapmandu** [comment]
Perhaps back-porting this may help..

https://github.com/cfwheels/cfwheels/commit/ae87f8991903a90a520b338ef105020c1a1b6809

**Comment by smilebringergames** [comment]
Actually as per below article, it could be that they fixed this in a maintenance update for 1.4.6. I might have to look into this myself as well: https://cfwheels.org/blog/page/4/

[CFWheels 1.4.6 maintenance Release](https://cfwheels.org/blog/cfwheels-1-4-6-maintenance-release/)
October 2nd, 2017 by Tom King

Today sees a maintenance release for the 1.4.x series. Please note that as 2.0 is now released, future updates to the 1.x branch will be limited to essential bug fixes only.

[Download 1.4.6 today](https://github.com/cfwheels/cfwheels/releases/download/v1.4.6/cfwheels.1.4.6.zip) to fix the following:

Bug Fixes
Made humanize() keep spaces in input.
Added spatial datatypes for MySQL.
Scope variable to avoid object being returned as NULL.
Include “MariaDB” in database check connection string.
Fixes MySQL attempts to insert nulls for blank strings.
Tags: [1.4](https://cfwheels.org/blog/tag/1-4/)
Posted in [Releases](https://cfwheels.org/blog/category/releases/) | Comments Off
